### PR TITLE
OSSM-5871 Introduce IstioCNI resource and controller

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by luksa** [comment]
I forgot to mention that I'm waiting for #1640 to be merged so that I can then modify the test to use the new IstioCNI resource.

**Comment by dgn** [comment]
> I forgot to mention that I'm waiting for #1640 to be merged so that I can then modify the test to use the new IstioCNI resource.

I was about to ask that, thanks!

**Comment by luksa** [comment]
@dgn Test added.

File: api/v1alpha1/istiocni_types.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2023.
```
Comment (jwendell) [inline api/v1alpha1/istiocni_types.go]: remove the year and put Istio Authors?

**Review by jwendell — APPROVED** [review]
Please update README :)

File: api/v1alpha1/istiocni_types.go (hunk 1 above)
Comment (luksa) [reply api/v1alpha1/istiocni_types.go]: Done.

**Pull request merged**